pipeline/transform.py

```diff
@@ -0,0 +1,145 @@
+"""Script for transforming data from the Bandcamp API"""
+import logging
+import re
+from datetime import datetime
+import requests
+import pandas as pd
+from bs4 import BeautifulSoup
+from extract import main_extract
+
+
+def config_log() -> None:
+    """Terminal logs configuration"""
+    logging.basicConfig(
+        format="{asctime} - {levelname} - {message}",
+        style="{",
+        datefmt="%Y-%m-%d %H:%M",
+        level=logging.INFO,
+    )
+
+
+def convert_from_unix_to_datetime(unix: str) -> datetime:
+    """Converts a unix timestamp to a regularly
+     formatted date"""
+
+    date = datetime.fromtimestamp(unix)
+
+    return date
+
+
+def convert_to_full_url(url: str) -> str:
+    """Checks if a url is valid and returns
+    the full url if it is not"""
+
+    if url.startswith("http"):
+        return url
+
+    return "https:" + url
+
+
+def get_sale_information(sales_dict: dict) -> list[dict]:
+    """Returns a list of dictionaries containing the
+    sales information"""
+    sales_info = []
+
+    events = sales_dict["feed_data"]["events"]
+
+    for event in events:
+        items = event["items"][0]
+
+        if items["item_type"] not in ["a", "t"]:
+            continue
+
+        if items["url"]:
+            release_date = get_release_date(items["url"])
+            genres = get_genres_from_url(items["url"])
+        else:
+            release_date = "None"
+            genres = []
+
+        sale_date = convert_from_unix_to_datetime(items["utc_date"])
+
+        event_information = {"item_type": items["item_type"],
+                             "item_description": items["item_description"],
+                             "album_title": items["album_title"],
+                             "artist_name": items["artist_name"],
+                             "country": items["country"],
+                             "amount_paid_usd": items["amount_paid_usd"],
+                             "genres": genres,
+                             "release_data": release_date,
+                             "sale_date": sale_date
+                             }
+
+        sales_info.append(event_information)
+
+    return sales_info
+
+
+def get_genres_from_url(artist_url: str) -> list[str]:
+    """Gets the lists of genres from a given artist url.
+        Returns an empty list if none can be found."""
```

Comment:
Could you add a try, except block here as currently, if the API or web scraping fails, the entire pipeline could break. Add exception handling to ensure robustness.